src/Sylius/Component/Product/Repository/ProductVariantRepositoryInterface.php

```diff
@@ -66,4 +66,6 @@ public function findByPhrase(string $phrase, string $locale, ?int $limit = null)
      * @return array|string[]
      */
     public function getCodesOfAllVariants(): array;
+
+    public function findDefaultVariant(ProductInterface $product, bool $enabled = true): ?ProductVariantInterface;
```

Comment:
If you don't want to break BC then you can add the position sorting in the `findOneBy` call that was there before

Reply:
Can you please elaborate on the nature of BC, I'm not following.

Reply:
I think it's better to have custom methods instead of using methods like `findOneBy` because it makes refactoring and customizations easier.
But perhaps a major release is better for this purpose.

Reply:
You are right, reverting it.

Reply:
The `enabled` parameter bothers me a bit. I don't see why would it ever be `false`.
It should rather be something like `onlyEnabled` and if it's `true` then conditionally add the `where o.enabled = true` clause.
Otherwise, there's no way to get the default variant regardless of its status.

Reply:
Oh, never mind, you're reverting it.

Reply:
@vvasiloi To answer your question, I named the repository without "Enabled" word so it can be used in a shop or admin context without overriding this method. It was just to prepare some hypothetic futur feature like we discussed 😉

Reply:
Custom `findDefaultVariant` method was much better, I still don't understand about BC.

It was introduced in previous minor version -- it cannot be considered BC unless you will remove repository class entirely.
If somebody extended this class in previous version, then it will be still working as is.

Voting to revert to `findDefaultVariant`